### PR TITLE
[SPARK-11814][Streaming]Add better default checkpoint duration

### PR DESCRIPTION
**Pull request by tdas**
DStream checkpoint interval is by default set at max(10 second, batch interval). That's bad for large batch intervals where the checkpoint interval = batch interval, and RDDs get checkpointed every batch.
This PR is to set the checkpoint interval of trackStateByKey to 10 \* batch duration.

### CODE REVIEW

**Comment by tdas** [comment]
@zsxwing Please take a look.

**Comment by zsxwing** [comment]
I just noticed `TrackStateByKeySuite` leaks a SparkContext. Could you fix it?

File: streaming/src/test/scala/org/apache/spark/streaming/TrackStateByKeySuite.scala [hunk 1]
```diff
@@ -436,6 +437,41 @@ class TrackStateByKeySuite extends SparkFunSuite with BeforeAndAfterAll with Bef
...
+  test("trackStateByKey - checkpoint durations") {
+    val privateMethod = PrivateMethod[InternalTrackStateDStream[_, _, _, _]]('internalStream)
+
+    def testCheckpointDuration(
+        batchDuration: Duration,
+        expectedCheckpointDuration: Duration,
+        explicitCheckpointDuration: Option[Duration] = None
+      ): Unit = {
+      try {
+        ssc = new StreamingContext(sc, batchDuration)
```
Comment (zsxwing) [inline streaming/src/test/scala/org/apache/spark/streaming/TrackStateByKeySuite.scala]: This one will create a new `StreamingContext` and set it as the active StreamingContext. But the previous one created in `before` won't be stopped?

Comment (tdas) [inline streaming/src/test/scala/org/apache/spark/streaming/TrackStateByKeySuite.scala]: The absolute first one created in the `before {...}` is not started, so there is nothing to stop. And the context start in the function will be stopped by the finally.

Comment (zsxwing) [inline streaming/src/test/scala/org/apache/spark/streaming/TrackStateByKeySuite.scala]: Oh, right. Sorry for the false alarm.

**Comment by tdas** [comment]
That's funny! How come the leak did not cause downstream failures?

**Comment by zsxwing** [comment]
LGTM

**Comment by tdas** [comment]
Thanks @zsxwing , merging this to master and 1.6

**Pull request closed without merging**